ppl/zqd/ingest/log.go

```diff
@@ -91,7 +92,13 @@ func NewLogOp(ctx context.Context, store storage.Storage, req api.LogPostRequest
 }
 
 func (p *LogOp) Warn(msg string) error {
-	p.warnings = append(p.warnings, msg)
+	// warnings received before we've started our goroutine are
+	// saved here and will be drained in start()
+	if p.warnings != nil {
+		p.warnings = append(p.warnings, msg)
```

Comment:
I'm not clear on whether this could ever be called from a goroutine, but if it is, then writing to the channel will be thread-safe, but the append will not be thread-safe without a lock.

Reply:
Yes, the only goroutine that calls Warn is the one started by NewLogOp (`go p.start(ctx, store)`). This is pre-existing behavior. I'll add a comment to clarify.